src/ext.rs

```diff
@@ -832,6 +858,31 @@ fn timeout2ms(dur: DWORD) -> Option<u32> {
 }
 
 #[cfg(any(feature = "nightly", feature = "duration"))]
+fn linger2dur(linger_opt: linger) -> Option<Duration> {
+    let dur = Duration::from_secs(linger_opt.l_linger as u64);
+    if dur.as_secs() == 0 {
+        None
+    }
+    else {
+        Some(dur)
+    }
+}
+
+#[cfg(any(feature = "nightly", feature = "duration"))]
+fn dur2linger(dur: Option<Duration>) -> linger {
+    match dur {
+        Some(d) => {
+            let onoff = if d.as_secs() == 0 { 0 } else { 1 };
+            linger {
+                l_onoff: onoff as u16,
```

Comment:
Hm are you sure about that documentation? It looks like the nonzero-ness is only referred to when talking about `l_onoff`. I'm not sure if `Some(Duration::new(0))` actually makes sense, but if we can communicate it to the system we should.